main.py

```diff
@@ -2,19 +2,33 @@
 
 import yaml
 
+# DBConnect Session
+from databricks.connect import DatabricksSession
+
 from src.hotel_reservation.data_processor import DataProcessor
+from src.hotel_reservation.reservation_model import ReservationModel
+from src.hotel_reservation.utils import plot_feature_importance, visualize_results
+
+spark = DatabricksSession.builder.profile("adb-1846957892648178").getOrCreate()
```

Comment:
please use SparkSession instead. It works as well in VS code when Databricks extension is enabled. The approach you use is not great: it only runs on your machine.